src/repository/Repository.ts

```diff
@@ -345,5 +345,20 @@ export class Repository<Entity extends ObjectLiteral> {
     async clear(): Promise<void> {
         return this.manager.clear(this.metadata.target);
     }
-
+    
+    /**
+     * Reloads the supplied entity into the latest state of the record from the database
+     */
+    reload(entity: Entity): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            const newEntity = await this.findOneById(entity.id);
+            
+            if (!newEntity) {
+                reject("An error ocurred while reloading entity");
+            }
+            
+            Object.assign(entity, newEntity);
+            resolve();
```

Comment:
Why not resolve newEntity? Why mutate argument? In my opinion `Object.assign` should be in BaseEntity.

Reply:
I hadn't thought about it. Makes more sense to assign it inside BaseEntity

Reply:
TBH I don't find `reload` method useful in repository, so I think the whole method body must be moved into BaseEntity

Reply:
Ok, sounds good to me. I'll update 👍